internal/releasesources/release_sources.go

```diff
@@ -0,0 +1,510 @@
+			dynamicRelease.Default = false
+		}
+		mergedReleaseImages = append(mergedReleaseImages, dynamicRelease)
+	}
+
+	return mergedReleaseImages
+}
+
+func (h *releaseSourcesHandler) syncReleaseImagesWithErr() error {
+	err := h.validateReleaseSources()
+	if err != nil {
+		return err
+	}
+
+	err = h.validateStaticReleaseImages()
+	if err != nil {
+		return err
+	}
+	dynamicDBReleaseImages, err := h.getDynamicReleaseImages()
+	if err != nil {
+		return err
+	}
+	h.log.Debugf("Found %d dynamic release Images", len(dynamicDBReleaseImages))
+	supportLevels, err := h.getSupportLevels()
+	if err != nil {
+		return err
+	}
+	err = h.setSupportLevels(dynamicDBReleaseImages, supportLevels)
+	if err != nil {
+		return err
+	}
+	staticDBReleaseImages := h.getStaticReleaseImages()
+	h.log.Debugf("Found %d static release Images", len(staticDBReleaseImages))
+	err = h.setSupportLevels(staticDBReleaseImages, supportLevels)
+	if err != nil {
+		return err
+	}
+	err = h.setDefaultReleaseImage(dynamicDBReleaseImages)
+	if err != nil {
+		return err
+	}
+
+	dynamicDBReleaseImages, err = h.removeReleaseImageDuplicates(dynamicDBReleaseImages)
+	if err != nil {
+		return err
+	}
+
+	dbReleaseImages := h.mergeDBReleaseImages(staticDBReleaseImages, dynamicDBReleaseImages)
+
+	h.log.Debug("Starting SQL transaction")
+	tx := h.db.Begin()
+	// Deleting all releases before adding again in order to
+	// store only the releases according to RELEASE_SOURCES
+	err = h.deleteAllReleases(tx)
+	if err != nil {
+		tx.Rollback()
+		return err
+	}
+
+	err = h.insertToDB(tx, dbReleaseImages)
+	if err != nil {
+		tx.Rollback()
+		return err
+	}
+
+	h.log.Debug("Commiting changes")
+	return tx.Commit().Error
+}
+
+// getReleaseImagesMajorOCPVersions retrieves the major OCP versions for both static and dynamic release images discovered.
+func (h *releaseSourcesHandler) getReleaseImagesMajorOCPVersions() (ocpMajorVersionSet, error) {
+	majorVersions := ocpMajorVersionSet{}
+
+	// First from static release images
+	for _, releaseImage := range h.staticReleaseImages {
+		majorVersion, err := common.GetMajorVersion(*releaseImage.OpenshiftVersion)
+		if err != nil {
+			return nil, errors.Errorf("error occurred while trying to get the major version of %s", *releaseImage.OpenshiftVersion)
+		}
+		majorVersions[*majorVersion] = true
+	}
+
+	// Then from dynamic release images
+	for _, releaseSource := range h.releaseSources {
+		majorVersion, err := common.GetMajorVersion(*releaseSource.OpenshiftVersion)
+		if err != nil {
+			return nil, errors.Errorf("error occurred while trying to get the major version of %s", *releaseSource.OpenshiftVersion)
+		}
+		majorVersions[*majorVersion] = true
+	}
+
+	return majorVersions, nil
+}
+
+// getSupportLevels retrieves a mapping from OCP major.minor versions
+// to their corresponding support levels for both static and dynamic release images.
+func (h *releaseSourcesHandler) getSupportLevels() (ocpVersionSupportLevels, error) {
```

Comment:
interesting how many api calls we will produce each run? Afraid of ddos of ocm a little bit

Reply:
for - `OCM`, for example if we have this `ReleaseSources`:

```
var releaseSources = models.ReleaseSources{
	{
		OpenshiftVersion: swag.String("4.10"),
		UpgradeChannels: []*models.UpgradeChannel{
			{
				CPUArchitecture: swag.String(common.X86CPUArchitecture),
				Channels:        []string{OpenshiftReleaseChannelStable, OpenshiftReleaseChannelCandidate},
			},
                        {
				CPUArchitecture: swag.String(common.ARM64CPUArchitecture),
				Channels:        []string{OpenshiftReleaseChannelStable, OpenshiftReleaseChannelCandidate},
			},
		},
	},
        .
        .
        .
}
```

we will have an API call for each combination so -
```
4.10, x86_64, stable
4.10, x86_64, candidate
4.10, ARM64CPUArchitecture, stable,
4.10, ARM64CPUArchitecture, candidate,
``` 

So assuming we collect `(x86_64, arm64, aarch64, s390x, ppcle64, multi)`, from 2 channels `(candidate, stable)`, for about 4 versions each time (2 production, 2 maintenance), we should sum up to about
6 * 2 * 4 = 48 API calls every run, (every 30 min) I think we should be good.

Reply:
Also I think every connected OCP cluster in the world uses these endpoints. Our few deployments of our service will be a drop in the bucket.